vendor/api/v1/views.py

```diff
@@ -0,0 +1,191 @@
+from django.apps import apps
+from django.contrib import messages
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpResponse
+from django.shortcuts import redirect, get_object_or_404
+from django.urls import reverse_lazy, reverse
+from django.utils.translation import gettext_lazy as _
+from django.utils import timezone
+from django.views import View
+
+from vendor.config import VENDOR_PRODUCT_MODEL
+from vendor.models import CustomerProfile, Invoice, Offer, Receipt
+from vendor.processors import get_site_payment_processor
+from vendor.utils import get_or_create_session_cart, get_site_from_request
+
+Product = apps.get_model(VENDOR_PRODUCT_MODEL)
+class VendorIndexAPI(View):
+    """
+    docstring
+    """
+    def get(self, request, *args, **kwargs):
+        print('index api')
+        return HttpResponse('<h1>Welcome to vendor APIs<h1>')
+
+
+class AddToCartView(View):
+    '''
+    Create an order item and adds it to the order. No login required as the request can come
+    from a user without a session.
+    '''
+    def session_cart(self, request, offer):
+        offer_key = str(offer.pk)
+        session_cart = get_or_create_session_cart(request.session)
+
+        if offer_key not in session_cart:
+            session_cart[offer_key] = {}
+            session_cart[offer_key]['quantity'] = 0
+
+        session_cart[offer_key]['quantity'] += 1
+
+        if not offer.allow_multiple:
+            session_cart[offer_key]['quantity'] = 1
+
+        return session_cart
+
+    def post(self, request, *args, **kwargs):
+        try:
+            offer = Offer.objects.get(site=get_site_from_request(request), slug=self.kwargs["slug"], available=True)
+        except ObjectDoesNotExist:
+            messages.error(_("Offer does not exist or is unavailable"))
+            return redirect('vendor:cart')
+
+        if request.user.is_anonymous:
```

Comment:
according to django docs, `request.user.is_authenticated` is the preferred way to check login status; not a blocker though